test_conformance/images/kernel_read_write/test_cl_ext_image_from_buffer.cpp

```diff
@@ -638,127 +676,149 @@ int image_from_buffer_fill_positive(cl_device_id device, cl_context context,
         CL_MEM_OBJECT_IMAGE1D_ARRAY, CL_MEM_OBJECT_IMAGE2D_ARRAY
     };
 
-    for (auto imageType : imageTypes)
+    std::vector<cl_mem_flags> flagTypes{
+        CL_MEM_READ_ONLY,
+        CL_MEM_WRITE_ONLY,
+        CL_MEM_READ_WRITE,
+        CL_MEM_KERNEL_READ_AND_WRITE
```

Comment:
Are we sure that this flag list the way it is correct here? The arg `flag` is passed both to `get_format_list` (okay) and to `clCreateBuffer` (not okay). As far as the spec goes, even with the relevant ext updates still reads as:
> [CL_MEM_KERNEL_READ_AND_WRITE](https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_MEM_KERNEL_READ_AND_WRITE): This flag is only used by [clGetSupportedImageFormats](https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clGetSupportedImageFormats) to query image formats that may be both read from and written to by the same kernel instance. To create a memory object that may be read from and written to use [CL_MEM_READ_WRITE](https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_MEM_READ_WRITE).

Going by the spec, this flag should not be fed to `clCreateBuffer`. It seems to me that either the code here or the spec needs an update.